main/rendering/timings.ts

```diff
@@ -1,47 +1,109 @@
 import { BKChapter } from '../models/projectFormat.model';
-import { Timings, LineTiming } from '../models/timings.model';
+import { Timings, SegmentTiming, PhraseTiming } from '../models/timings.model';
 
 export function chapterFormatToTimings(chapter: BKChapter): Timings {
   const timings: Timings = [];
-  for (const segment of chapter.segments) {
-    const contentWords = segment.text.split(' ');
-    const lineTiming: LineTiming = {
+  for (const [index, segment] of chapter.segments.entries()) {
+    const contentWords = segment.text.split(' ').filter((w) => w);
+
+    // if the last timing data has start === end, then set end to the audio duration (end of chapter)
+    const audioLength = chapter.audio.files[0].length;
```

Comment:
Isn't `files[0]` assuming there is only one file. Is this a valid assumption here?

Reply:
I don't think SAB supports multiple audio files per chapter. And the `scripture-app-builder.ts` import only gets one audio file. But I changed it just in case.